### DISCUSSION TITLE
NewLine and Right Prompt

### DISCUSSION THREAD

**Discussion by JohnPolansky** [starship/starship#5435] [Q&A] [answered]
I'm trying to setup a prompt with left/right and input is on a newline with no characters (I like being able to triple click and select the whole command without having `>%` or some other character messing it up. So i'm trying to set this up:

```
| left                                                                          right |
| _ <-Cursor                                                                          |
```

I simplified my config down to the basics to avoid any conflicts but I'm not finding any combination of `\r\n` that will work for what i want. If I don't use the right prompt it's easy enough:
```
format = "left\n\r"
```
but when I and add the `right_format` everything  the `right` ends up on the 2nd line. I can't seem to get it right. It seems like it should be simple, and I'm missing something
```
format = "left"
right_format = "right"
```
Thanks in advance!

**Comment by davidkna** [comment, marked as answer]
To right-align text on the first lines in a multi-line prompt, you can use the `$fill` module.

**Reply by JohnPolansky** [reply]
Wow I would not have expected that to work but it did Thanks!

```
format = "$directory$kubernetes$git_branch$fill$cmd_duration$status\n\r"

[fill]
symbol = ' '
```

**Comment by kaichevannes** [comment]
For others who want to keep the default prompt elements but move some elements to the right:
```toml
format = "$all$fill$cmd_duration$status$line_break$character"

[fill]
symbol = ' '
```

From the [docs](https://starship.rs/config/#default-prompt-format):
>"If you just want to extend the default format, you can use $all; modules you explicitly add to the format will not be duplicated."
